source/extensions/tracers/zipkin/zipkin_tracer_impl.cc

```diff
@@ -30,20 +30,41 @@ void ZipkinSpan::setTag(const std::string& name, const std::string& value) {
 
 void ZipkinSpan::injectContext(Http::HeaderMap& request_headers) {
   // Set the trace-id and span-id headers properly, based on the newly-created span structure.
-  request_headers.addReferenceKey(ZipkinCoreConstants::get().X_B3_TRACE_ID,
```

Comment:
Pretty sure if you just change `addReferenceKey` to `setReferenceKey` it will do what you want without the extra logic here? Same elsewhere?

Reply:
Sorry, I didn't notice this function, it is satisfied with the demand.